tools/crm_mon.c

```diff
@@ -729,6 +736,10 @@ mon_shutdown(int nsig)
 #if CURSES_ENABLED
 static sighandler_t ncurses_winch_handler;
 
+/* Signal handler installed the regular way (not into the main loop) for when
+ * the screen is resized.  Commonly, this happens when running in an xterm and
+ * the user changes its size.
+ */
```

Comment:
This makes me realize ncurses_winch_handler and refresh_trigger should be declared volatile

Reply:
Doing so makes gcc want mainloop_set_trigger to be defined to take a volatile.  Can we make that change in public API?

Reply:
Bah. I'm not sure whether that would implicitly cast or not for existing callers. But it seems like the sort of thing that doesn't belong in a library API except in unusual circumstances.

We could cast the volatile away when calling mainloop_set_trigger(), and let it be vulnerable for the duration of that call -- that function is incredibly simple, it would just be a handful of cycles. That would at least protect it within crm_mon.

As an aside, the mainloop API should never have been public ...

Reply:
We could also add a new mainloop_set_trigger function that is private API and takes the proper type.

Reply:
Added to #2288.